### PR TITLE
Fix vnet naming error for sap_system

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
VNet naming error for resource naming

## Solution
Logic error when calculating the vnet name

## Tests
<Please provide steps to test the PR>

## Notes
Solves https://github.com/Azure/sap-hana/issues/753

### CODE REVIEW

**Review by YunziZhangAzure — APPROVED** [review]

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf [hunk 1]
```diff
@@ -74,7 +74,7 @@ locals {
   vnet_sap_name   = local.vnet_sap_exists ? try(split("/", local.vnet_sap_arm_id)[8], "") : try(local.var_vnet_sap.name, "sap")
   vnet_nr_parts   = length(split("-", local.vnet_sap_name))
   // Default naming of vnet has multiple parts. Taking the second-last part as the name 
-  vnet_sap_name_prefix = try(substr(upper(local.vnet_sap_name), -5, 5), "") == "-VNET" ? substr(local.vnet_sap_name, 0, length(local.vnet_sap_name) - 5) : local.vnet_sap_name
+  vnet_sap_name_prefix = try(substr(upper(local.vnet_sap_name), -5, 5), "") == "-VNET" ? substr(split("-", local.vnet_sap_name)[(local.vnet_nr_parts -2)],0,7) : local.vnet_sap_name
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf:77]: I'm not sure why you need substr to cut 7 CHAR again although it will also handled [here](https://github.com/Azure/sap-hana/blob/65ac977ab1bf26c942c9c1aa26e62167bfc0452b/deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf#L66).

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf [hunk 2]
```diff
@@ -193,13 +193,13 @@ locals {
   customer_provided_names = try(local.anydb.dbnodes[0].name, "") == "" ? false : true
 
   dbnodes = flatten([[for idx, dbnode in try(local.anydb.dbnodes, [{}]) : {
-    name      = try("${dbnode.name}-0", format("%sd%s%03d%s%d%s", local.sid, local.anydb_sid, idx, local.anydb_oscode, idx, substr(var.random-id.hex, 0, 3))),
+    name      = try("${dbnode.name}-0", format("%sd%s%03d%s%d%s", lower(local.sid), local.anydb_sid, idx, local.anydb_oscode, idx, substr(var.random-id.hex, 0, 3))),
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf:196]: There is lower for SID when assign computer name [here](https://github.com/Azure/sap-hana/blob/65ac977ab1bf26c942c9c1aa26e62167bfc0452b/deploy/terraform/terraform-units/modules/sap_system/anydb_node/vm-anydb.tf#L107).
What is the purpose of this change?

**Review by rukawata — COMMENTED** [review]
Thanks for the PR!
Since beta branch is for stabilized pre-release version, please always create PR against feature branch.
We will merge them into beta once we get several fixes.

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf (hunk 1 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf:77]: It can probably be removed - however all of these will be superseded by the changes introduced by the naming module

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf (hunk 2 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/anydb_node/variables_local.tf:196]: this can be removed too - like above all these will be superseded by the code in the naming module

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf [hunk 3]
```diff
@@ -75,12 +75,12 @@ locals {
...
-  vnet_sap_exists = try(local.var_vnet_sap.is_existing, false)
-  vnet_sap_arm_id = local.vnet_sap_exists ? try(local.var_vnet_sap.arm_id, "") : ""
-  vnet_sap_name   = local.vnet_sap_exists ? try(split("/", local.vnet_sap_arm_id)[8], "") : try(local.var_vnet_sap.name, "sap")
+  vnet_sap_arm_id = try(local.var_vnet_sap.arm_id, "")
+  vnet_sap_exists = length(local.vnet_sap_arm_id) > 0 ? true : false
+  vnet_sap_name   = local.vnet_sap_exists ? split("/", local.vnet_sap_arm_id)[8] : try(local.var_vnet_sap.name, "")
   vnet_nr_parts   = length(split("-", local.vnet_sap_name))
   // Default naming of vnet has multiple parts. Taking the second-last part as the name 
-  vnet_sap_name_prefix = substr(try(substr(upper(local.vnet_sap_name), -5, 5), "") == "-VNET" ? substr(local.vnet_sap_name, 0, length(local.vnet_sap_name) - 5) : local.vnet_sap_name, 0, 7)
+  vnet_sap_name_prefix = try(substr(upper(local.vnet_sap_name), -5, 5), "") == "-VNET" ? substr(split("-", local.vnet_sap_name)[(local.vnet_nr_parts -2)],0,7) : local.vnet_sap_name
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf:83]: substr(xxx, 0, 7) is still here.

File: deploy/terraform/terraform-units/modules/sap_system/hdb_node/variables_local.tf [hunk 4]
```diff
@@ -84,7 +84,7 @@ locals {
   vnet_sap_name   = local.vnet_sap_exists ? try(split("/", local.vnet_sap_arm_id)[8], "") : try(local.var_vnet_sap.name, "")
   vnet_nr_parts   = length(split("-", local.vnet_sap_name))
   // Default naming of vnet has multiple parts. Taking the second-last part as the name 
-  vnet_sap_name_prefix = local.vnet_nr_parts >= 3 ? split("-", upper(local.vnet_sap_name))[local.vnet_nr_parts - 1] == "VNET" ? split("-", local.vnet_sap_name)[local.vnet_nr_parts - 2] : local.vnet_sap_name : local.vnet_sap_name
+  vnet_sap_name_prefix = try(substr(upper(local.vnet_sap_name), -5, 5), "") == "-VNET" ? substr(split("-", local.vnet_sap_name)[(local.vnet_nr_parts -2)],0,7) : local.vnet_sap_name
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/hdb_node/variables_local.tf:87]: Same as above.

**Review by rukawata — COMMENTED** [review]
Thanks for removing the substr.
There are two more places using this, please check.

**Review by rukawata — APPROVED** [review]
lgtm!
